src/front/app/components/globals/Input.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const classNames = require('classnames');
+
+const inputTypes = [
+  // Input Based
+  'text',
+  'email',
+  'password',
+  'number',
+  'date',
+  'datetime',
+  'datetime-local',
+  'search',
+  'tel',
+  'color',
+  'file',
+  // Select Based
+  'select',
+  // Text Area Based
+  'textarea',
+];
+
+class Input extends Component {
+  static propTypes = {
+    inputType: PropTypes.oneOf(inputTypes).isRequired,
+    name: PropTypes.string.isRequired,
+    id: PropTypes.string,
+    className: PropTypes.string,
+    children: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.node,
+      PropTypes.instanceOf(Component),
+    ]),
+    rowCount: PropTypes.number,
+  };
+
+  static defaultProps = {
+    inputType: 'text',
+  };
+
+  render() {
+    const {
+      inputType,
+      name,
+      id,
+      className,
+      children,
+      rowCount,
+      ...props
+    } = this.props;
+
+    const inputClass = classNames({
+      'form-control': inputType !== 'file',
+      'form-control-file': inputType === 'file',
+      [className]: !!className,
+    });
+
+    if (inputType === 'select' || children) {
+      return (
+        <select name={name} className={inputClass} {...props}>
+          {children}
+        </select>
+      );
+    }
+
+    if (inputType === 'textarea' || rowCount) {
+      return (
+        <textarea id={id || `#${name}`} name={name} className={inputClass} rows={rowCount} {...props} />
+      );
+    }
+
+    return (
+      <input id={id || `#${name}`} name={name} type={inputType} className={inputClass} {...props} />
```

Comment:
defaulting to name is not a bad idea, but not too in love with it, as it will get you used to it :thinking:

Reply:
I AM used to it. Laravel Forms work this way. I'm not sure why this would be a problem, though I'm wiling to change it if you consider it to be so. :thinking:

Reply:
If so, leave it as is and ignore :+1: